diff -ur parole-0.2.0.6.orig/configure.ac parole-0.2.0.6/configure.ac
--- parole-0.2.0.6.orig/configure.ac	2011-04-17 13:54:13.000000000 +0200
+++ parole-0.2.0.6/configure.ac	2011-09-28 13:38:01.415723354 +0200
@@ -78,7 +78,7 @@
 m4_define([dbus_minimum_version], [0.60])
 m4_define([dbus_glib_minimum_version], [0.70])
 
-m4_define([libxfcegui4_minimum_version],[4.6.0])
+m4_define([libxfce4ui_minimum_version],[4.8.0])
 m4_define([libxfce4util_minimum_version],[4.6.0])
 
 m4_define([libnotify_minimum_version], [0.4.1])
@@ -98,7 +98,7 @@
 XDT_CHECK_PACKAGE([DBUS], [dbus-1], [dbus_minimum_version])
 XDT_CHECK_PACKAGE([DBUS_GLIB], [dbus-glib-1], [dbus_glib_minimum_version])
 
-XDT_CHECK_PACKAGE([LIBXFCE4GUI], [libxfcegui4-1.0],[libxfcegui4_minimum_version])
+XDT_CHECK_PACKAGE([LIBXFCE4UI], [libxfce4ui-1],[libxfce4ui_minimum_version])
 XDT_CHECK_PACKAGE([LIBXFCE4UTIL], [libxfce4util-1.0],[libxfce4util_minimum_version])
 
 #=======================================================#
diff -ur parole-0.2.0.6.orig/gst/Makefile.am parole-0.2.0.6/gst/Makefile.am
--- parole-0.2.0.6.orig/gst/Makefile.am	2011-04-17 13:53:19.000000000 +0200
+++ parole-0.2.0.6/gst/Makefile.am	2011-09-28 13:38:01.415723354 +0200
@@ -23,10 +23,11 @@
 	parole-gst-iface.h
 	
 libparolegst_la_CFLAGS =		\
+	$(GTK_CFLAGS)             \
 	$(GST_VIDEO_CFLAGS)             \
         $(GST_INTERFACES_CFLAGS)	\
 	$(GTHREAD_CFLAGS)               \
-	$(LIBXFCE4GUI_CFLAGS)
+	$(LIBXFCE4UI_CFLAGS)
 
 libparolegst_la_LIBADD =		\
 	$(top_builddir)/parole/libparole.la\
diff -ur parole-0.2.0.6.orig/gst/parole-gst.c parole-0.2.0.6/gst/parole-gst.c
--- parole-0.2.0.6.orig/gst/parole-gst.c	2011-04-17 13:53:19.000000000 +0200
+++ parole-0.2.0.6/gst/parole-gst.c	2011-09-28 13:38:01.415723354 +0200
@@ -37,7 +37,7 @@
 #undef XFCE_DISABLE_DEPRECATED
 #endif
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include <gdk/gdkx.h>
 
@@ -1523,7 +1523,7 @@
     if ( gst->priv->state != gst->priv->target )
     {
 	gboolean ret_val = 
-	    xfce_confirm (_("The stream is taking too much time to load"), GTK_STOCK_OK, _("Stop"));
+	    xfce_dialog_confirm(NULL, GTK_STOCK_OK, _("Stop"), NULL, _("The stream is taking too much time to load"));
 	    
 	if ( ret_val )
 	{
@@ -1712,7 +1712,7 @@
  
     if ( G_UNLIKELY (gst->priv->playbin == NULL) )
     {
-	xfce_err (_("Unable to load playbin GStreamer plugin"
+	xfce_dialog_show_error (NULL, NULL, _("Unable to load playbin GStreamer plugin"
 		    ", check your GStreamer installation"));
 		    
 	g_error ("playbin load failed");
@@ -1732,7 +1732,7 @@
 	
 	if ( G_UNLIKELY (gst->priv->video_sink == NULL) )
 	{
-	    xfce_err (_("Unable to load video GStreamer plugin"
+	    xfce_dialog_show_error (NULL, NULL, _("Unable to load video GStreamer plugin"
 		      ", check your GStreamer installation"));
 	    g_error ("ximagesink load failed");
 	}
diff -ur parole-0.2.0.6.orig/plugins/tray/Makefile.am parole-0.2.0.6/plugins/tray/Makefile.am
--- parole-0.2.0.6.orig/plugins/tray/Makefile.am	2011-04-17 13:53:19.000000000 +0200
+++ parole-0.2.0.6/plugins/tray/Makefile.am	2011-09-28 13:38:01.419056688 +0200
@@ -19,7 +19,7 @@
 tray_icon_la_CFLAGS =				\
 	$(PLATFORM_CFLAGS)			\
 	$(GTK_CFLAGS)				\
-	$(LIBXFCE4GUI_CFLAGS)			\
+	$(LIBXFCE4UI_CFLAGS)			\
 	$(LIBXFCE4UTIL_CFLAGS)			\
 	$(LIBNOTIFY_CFLAGS)
 
@@ -44,4 +44,4 @@
 	$(desktop_in_files)
     
 DISTCLEANFILES = 		\
-	$(desktop_DATA)
\ No newline at end of file
+	$(desktop_DATA)
diff -ur parole-0.2.0.6.orig/plugins/tray/tray-provider.c parole-0.2.0.6/plugins/tray/tray-provider.c
--- parole-0.2.0.6.orig/plugins/tray/tray-provider.c	2011-04-17 13:53:19.000000000 +0200
+++ parole-0.2.0.6/plugins/tray/tray-provider.c	2011-09-28 13:39:06.435726288 +0200
@@ -33,7 +33,7 @@
 
 
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include "tray-provider.h"
 
diff -ur parole-0.2.0.6.orig/src/main.c parole-0.2.0.6/src/main.c
--- parole-0.2.0.6.orig/src/main.c	2011-04-17 13:53:19.000000000 +0200
+++ parole-0.2.0.6/src/main.c	2011-09-28 13:38:01.419056688 +0200
@@ -313,12 +313,8 @@
     {
 	builder = parole_builder_get_main_interface ();
 	parole_dbus_register_name (PAROLE_DBUS_NAME);
-	session = parole_session_get ();
+	session = parole_session_get (client_id);
 	
-	if ( client_id )
-	    parole_session_set_client_id (session, client_id);
-	    
-	parole_session_real_init (session);
 	player = parole_player_new ();
 	
 	if (fullscreen)
diff -ur parole-0.2.0.6.orig/src/Makefile.am parole-0.2.0.6/src/Makefile.am
--- parole-0.2.0.6.orig/src/Makefile.am	2011-04-17 13:53:19.000000000 +0200
+++ parole-0.2.0.6/src/Makefile.am	2011-09-28 13:38:01.419056688 +0200
@@ -13,11 +13,12 @@
 	-DPAROLE_PLUGINS_DIR=\"$(libdir)/parole-$(PAROLE_VERSION_API)\"\
 	-DPAROLE_PLUGINS_DATA_DIR=\"$(datadir)/parole/parole-plugins-$(PAROLE_VERSION_API)\"\
 	-DPAROLE_COMPILATION			\
+        $(GTK_CFLAGS)                       \
         $(GTHREAD_CFLAGS)                       \
 	$(GIO_CFLAGS)				\
 	$(GMODULE_CFLAGS)			\
         $(DBUS_GLIB_CFLAGS)                     \
-        $(LIBXFCE4GUI_CFLAGS)                   \
+        $(LIBXFCE4UI_CFLAGS)                   \
 	$(GST_VIDEO_CFLAGS)			\
 	$(GST_INTERFACES_CFLAGS)		\
 	$(TAGLIBC_CFLAGS)
@@ -78,11 +79,12 @@
 	$(top_builddir)/gst/libparolegst.la 	\
 	$(top_builddir)/common/libparolecommon.la\
 	$(top_builddir)/common/libparolescreensaver.la\
+        $(GTK_LIBS)                         \
         $(GTHREAD_LIBS)                         \
 	$(GIO_LIBS)				\
 	$(GMODULE_LIBS)				\
         $(DBUS_GLIB_LIBS)                       \
-        $(LIBXFCE4GUI_LIBS)                     \
+        $(LIBXFCE4UI_LIBS)                     \
 	$(GST_VIDEO_LIBS)			\
 	$(GST_INTERFACES_LIBS)			\
 	$(TAGLIBC_LIBS)
diff -ur parole-0.2.0.6.orig/src/parole-about.c parole-0.2.0.6/src/parole-about.c
--- parole-0.2.0.6.orig/src/parole-about.c	2011-04-17 13:53:19.000000000 +0200
+++ parole-0.2.0.6/src/parole-about.c	2011-09-28 13:42:35.262402377 +0200
@@ -31,73 +31,56 @@
 #undef XFCE_DISABLE_DEPRECATED
 #endif
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include "parole-about.h"
 #include "parole-utils.h"
 
 void  parole_about (void)
 {
-    XfceAboutInfo *info;
-    GtkWidget *dialog;
     gint x, y;
     GdkPixbuf *icon;
-    guint n;
 
-    static const struct
-    {
-	gchar *name, *email, *language;
-    } 	
-    translators[] = 
-    {
-	{"astur", "malditoastur@gmail.com", "ast",},
-	{"Carles Muñoz Gorriz", "carlesmu@internautas.org", "ca.po",},
-	{"Per Kongstad", "p_kongstad@op.pl", "da.po",},
-	{"Christoph Wickert", "cwickert@fedoraproject.org", "de.po",},
-	{"elega", "elega@elega.com.ar","es",},
-	{"Piarres Beobide", "pi+debian@beobide.net", "eu",},
-	{"Douart Patrick", "patrick.2@laposte.net", "fr",},
-	{"Leandro Regueiro", "leandro.regueiro@gmail.com", "gl",},
-	{"Andhika Padmawan", "andhika.padmawan@gmail.com", "id",},
-	{"Masato Hashimoto", "cabezon.hashimoto@gmail.com", "ja",},
-	{"Rihards Prieditis", "rprieditis@gmail.com", "lv",},
-	{"Mario Blättermann", "mariobl@gnome.org", "nl",},
-	{"Sérgio Marques", "smarquespt@gmail.com", "pt",},
-	{"Vlad Vasilev", "lortwer@gmail.com", "ru",},
-	{"Robert Hartl", "hartl.robert@gmail.com", "sk",},
-	{"Samed Beyribey", "ras0ir@eventualis.org", "tr",},
-	{"Motsyo Gennadi", "drool@altlinux.ru", "uk",},
-	{"Hunt Xu", "huntxu@live.cn", "zh_CN",},
-	
-    };
-
-    info = xfce_about_info_new ("Parole", VERSION, _("Parole Media Player"),
-                                XFCE_COPYRIGHT_TEXT ("2009-2011", "Ali Abdallah"), 
-				XFCE_LICENSE_GPL);
-
-    xfce_about_info_set_homepage (info, "http://goodies.xfce.org/projects/applications/parole");
-    xfce_about_info_add_credit (info, "Ali Abdallah", "aliov@xfce.org", _("Author/Maintainer"));
-  
-
-    for (n = 0; n < G_N_ELEMENTS (translators); ++n) 
-    {
-	gchar *s;
-	s = g_strdup_printf (_("Translator (%s)"), translators[n].language);
-	xfce_about_info_add_credit (info, translators[n].name, translators[n].email, s);
-	g_free (s);
-    }
+    static gchar *translators = 
+	"astur <malditoastur@gmail.com> ast\n"
+	"Carles Muñoz Gorriz <carlesmu@internautas.org> ca.po\n"
+	"Per Kongstad <p_kongstad@op.pl> da.po\n"
+	"Christoph Wickert <cwickert@fedoraproject.org> de.po\n"
+	"elega <elega@elega.com.ar> es\n"
+	"Piarres Beobide <pi+debian@beobide.net> eu\n"
+	"Douart Patrick <patrick.2@laposte.net> fr\n"
+	"Leandro Regueiro <leandro.regueiro@gmail.com> gl\n"
+	"Andhika Padmawan <andhika.padmawan@gmail.com> id\n"
+	"Masato Hashimoto <cabezon.hashimoto@gmail.com> ja\n"
+	"Rihards Prieditis <rprieditis@gmail.com> lv\n"
+	"Mario Blättermann <mariobl@gnome.org> nl\n"
+	"Sérgio Marques <smarquespt@gmail.com> pt\n"
+	"Vlad Vasilev <lortwer@gmail.com> ru\n"
+	"Robert Hartl <hartl.robert@gmail.com> sk\n"
+	"Samed Beyribey <ras0ir@eventualis.org> tr\n"
+	"Motsyo Gennadi <drool@altlinux.ru> uk\n"
+	"Hunt Xu <huntxu@live.cn> zh_CN\n";
+
+    const gchar *auth[] = { "Ali Abdallah <aliov@xfce.org> Author/Maintainer", NULL };
 
     gtk_icon_size_lookup (GTK_ICON_SIZE_DIALOG, &x, &y);
     icon = parole_icon_load ("parole", x);
-    
-    dialog = xfce_about_dialog_new_with_values (NULL, info, icon);
-    
-	
-    gtk_dialog_run (GTK_DIALOG (dialog));
-    gtk_widget_destroy (dialog);
-
-    xfce_about_info_free (info);
-    
-    if (icon)
+
+#if !GTK_CHECK_VERSION (2, 18, 0)
+    gtk_about_dialog_set_email_hook (exo_gtk_url_about_dialog_hook, NULL, NULL);
+    gtk_about_dialog_set_url_hook (exo_gtk_url_about_dialog_hook, NULL, NULL);
+#endif
+    gtk_show_about_dialog(NULL,
+       "logo", icon,
+       "program-name", "Parole",
+       "license", xfce_get_license_text (XFCE_LICENSE_TEXT_GPL),
+       "version", VERSION,
+       "comments", _("Parole Media Player"),
+       "website", "http://goodies.xfce.org/projects/applications/parole",
+       "copyright", _("Copyright (c) 2009-2011 Ali Abdallah\n"),
+       "authors", auth,
+       "translator-credits", translators, NULL);
+
+   if (icon)
 	g_object_unref (G_OBJECT (icon));
 }
diff -ur parole-0.2.0.6.orig/src/parole-player.c parole-0.2.0.6/src/parole-player.c
--- parole-0.2.0.6.orig/src/parole-player.c	2011-04-17 13:53:30.000000000 +0200
+++ parole-0.2.0.6/src/parole-player.c	2011-09-28 13:44:05.459073112 +0200
@@ -29,6 +29,7 @@
 #ifdef HAVE_XF86_KEYSYM
 #include <X11/XF86keysym.h>
 #endif
+#include <X11/Xatom.h>
 
 #include <gdk/gdkx.h>
 #include <gdk/gdkkeysyms.h>
@@ -38,7 +39,7 @@
 #undef XFCE_DISABLE_DEPRECATED
 #endif
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include <dbus/dbus-glib.h>
 
@@ -980,7 +981,7 @@
 static void
 parole_player_error_cb (ParoleGst *gst, const gchar *error, ParolePlayer *player)
 {
-    xfce_err ("%s", error);
+    xfce_dialog_show_error (NULL, NULL, "%s", error);
     parole_screen_saver_uninhibit (player->priv->screen_saver);
     parole_player_stopped (player);
 }
@@ -1702,13 +1703,6 @@
 #endif
 
 static void
-parole_player_session_die_cb (ParolePlayer *player)
-{
-    player->priv->exit = TRUE;
-    parole_gst_terminate (PAROLE_GST (player->priv->gst));
-}
-
-static void
 parole_gst_set_default_aspect_ratio (ParolePlayer *player, GtkBuilder *builder)
 {
     ParoleAspectRatio ratio;
@@ -1891,10 +1885,7 @@
     g_signal_connect_swapped (player->priv->conf, "notify::reset-saver",
 			      G_CALLBACK (parole_player_reset_saver_changed_cb), player);
     
-    player->priv->session = parole_session_get ();
-    
-    g_signal_connect_swapped (player->priv->session, "die",
-			      G_CALLBACK (parole_player_session_die_cb), player);
+    player->priv->session = parole_session_get (NULL);
     
     player->priv->gst = parole_gst_new (FALSE, player->priv->conf);
 
diff -ur parole-0.2.0.6.orig/src/parole-plugins-manager.c parole-0.2.0.6/src/parole-plugins-manager.c
--- parole-0.2.0.6.orig/src/parole-plugins-manager.c	2011-04-17 13:53:19.000000000 +0200
+++ parole-0.2.0.6/src/parole-plugins-manager.c	2011-09-28 13:38:01.422390022 +0200
@@ -33,7 +33,7 @@
 #undef XFCE_DISABLE_DEPRECATED
 #endif
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include <parole/parole-provider-plugin.h>
 
@@ -413,7 +413,7 @@
     /*No plugins found*/
     if ( manager->priv->array->len == 0 )
     {
-	xfce_info ("%s", _("No installed plugins found on this system")); 
+	xfce_dialog_show_info(NULL, NULL, "%s", _("No installed plugins found on this system")); 
 	return;
     }
     
diff -ur parole-0.2.0.6.orig/src/parole-session.c parole-0.2.0.6/src/parole-session.c
--- parole-0.2.0.6.orig/src/parole-session.c	2011-04-17 13:53:19.000000000 +0200
+++ parole-0.2.0.6/src/parole-session.c	2011-09-28 13:38:01.425723355 +0200
@@ -31,7 +31,7 @@
 #undef XFCE_DISABLE_DEPRECATED
 #endif
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include "parole-session.h"
 
@@ -42,13 +42,12 @@
 
 struct ParoleSessionPrivate
 {
-    SessionClient *client;
-    gboolean	   managed;
+    XfceSMClient  *client;
+    gboolean       managed;
 };
 
 enum
 {
-    DIE,
     SAVE_STATE,
     LAST_SIGNAL
 };
@@ -58,31 +57,10 @@
 G_DEFINE_TYPE (ParoleSession, parole_session, G_TYPE_OBJECT)
 
 static void
-parole_session_die (gpointer client_data)
-{
-    ParoleSession *session;
-    
-    session = parole_session_get ();
-    
-    if ( session->priv->managed )
-	g_signal_emit (G_OBJECT (session), signals [DIE], 0);
-}
-
-static void
 parole_session_class_init (ParoleSessionClass *klass)
 {
     GObjectClass *object_class = G_OBJECT_CLASS (klass);
 
-    signals [DIE] = 
-        g_signal_new ("die",
-                      PAROLE_TYPE_SESSION,
-                      G_SIGNAL_RUN_LAST,
-                      G_STRUCT_OFFSET(ParoleSessionClass, die),
-                      NULL, NULL,
-                      g_cclosure_marshal_VOID__VOID,
-                      G_TYPE_NONE, 0, G_TYPE_NONE);
-		      
-
     signals [SAVE_STATE] = 
         g_signal_new ("save-state",
                       PAROLE_TYPE_SESSION,
@@ -103,26 +81,6 @@
     session->priv = PAROLE_SESSION_GET_PRIVATE (session);
     
     session->priv->client = NULL;
-    
-    session->priv->client = client_session_new_full (NULL,
-						     SESSION_RESTART_NEVER,
-						     40,
-						     NULL,
-						     (gchar *) PACKAGE_NAME,
-						     NULL,
-						     NULL,
-						     NULL,
-						     NULL,
-						     NULL,
-						     NULL);
-						     
-    if ( G_UNLIKELY (session->priv->client == NULL ) )
-    {
-	g_warning ("Failed to connect to session manager");
-	return;
-    }
-    
-    session->priv->client->die     = parole_session_die;
 }
 
 static void
@@ -134,45 +92,52 @@
     
     if ( session->priv->client != NULL )
     {
-	client_session_free (session->priv->client);
+	g_object_unref (session->priv->client);
     }
 
     G_OBJECT_CLASS (parole_session_parent_class)->finalize (object);
 }
 
 ParoleSession *
-parole_session_get (void)
+parole_session_get (const gchar *client_id)
 {
     static gpointer parole_session_obj = NULL;
-    
+    ParoleSession *session = NULL;
+    GError *error = NULL;
+    gchar *current_dir;
+
+    const gchar *restart_command[] =
+    {
+        "parole",
+	"--restart",
+	NULL
+    };
+
     if ( G_LIKELY (parole_session_obj != NULL ) )
     {
 	g_object_ref (parole_session_obj);
     }
     else
     {
-	parole_session_obj = g_object_new (PAROLE_TYPE_SESSION, NULL);
+        session = g_object_new (PAROLE_TYPE_SESSION, NULL);
+        current_dir = g_get_current_dir ();
+        session->priv->client = xfce_sm_client_get_full (XFCE_SM_CLIENT_RESTART_NORMAL,
+                                                         XFCE_SM_CLIENT_PRIORITY_DEFAULT,
+	                                                 client_id,
+                                                         current_dir,
+                                                         restart_command,
+							 PACKAGE_NAME ".desktop");
+        g_free (current_dir);
+        session->priv->managed = xfce_sm_client_connect (session->priv->client, &error);
+
+        if ( G_UNLIKELY (error) )
+        {
+            g_warning ("Failed to connect to session manager : %s", error->message);
+            g_error_free (error);
+        }
+	parole_session_obj = session;
 	g_object_add_weak_pointer (parole_session_obj, &parole_session_obj);
     }
     
     return PAROLE_SESSION (parole_session_obj);
 }
-
-void parole_session_real_init       (ParoleSession *session)
-{
-    g_return_if_fail (PAROLE_IS_SESSION (session));
-    g_return_if_fail (session->priv->managed == FALSE);
-    
-    session->priv->managed = session_init (session->priv->client);
-}
-
-void parole_session_set_client_id (ParoleSession *session, const gchar *client_id)
-{
-    g_return_if_fail (PAROLE_IS_SESSION (session));
-    g_return_if_fail (session->priv->managed == FALSE);
-    
-    if ( G_UNLIKELY (session->priv->client == NULL) )
-	return;
-    
-    client_session_set_client_id (session->priv->client, client_id);
-}
diff -ur parole-0.2.0.6.orig/src/parole-session.h parole-0.2.0.6/src/parole-session.h
--- parole-0.2.0.6.orig/src/parole-session.h	2011-04-17 13:53:19.000000000 +0200
+++ parole-0.2.0.6/src/parole-session.h	2011-09-28 13:38:01.425723355 +0200
@@ -50,12 +50,8 @@
 
 GType        			parole_session_get_type        (void) G_GNUC_CONST;
 
-ParoleSession       	       *parole_session_get             (void);
+ParoleSession       	       *parole_session_get             (const gchar *client_id);
 
-void				parole_session_real_init       (ParoleSession *session);
-
-void 				parole_session_set_client_id   (ParoleSession *session, 
-								const gchar *client_id);
 G_END_DECLS
 
 #endif /* __PAROLE_SESSION_H */
diff -ur parole-0.2.0.6.orig/src/parole-statusbar.c parole-0.2.0.6/src/parole-statusbar.c
--- parole-0.2.0.6.orig/src/parole-statusbar.c	2011-04-17 13:53:19.000000000 +0200
+++ parole-0.2.0.6/src/parole-statusbar.c	2011-09-28 13:38:01.425723355 +0200
@@ -28,7 +28,7 @@
 
 #include <gtk/gtk.h>
 
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include "parole-builder.h"
 #include "parole-statusbar.h"
